lib/cabin/channel.rb

```diff
@@ -123,7 +140,7 @@ def subscribe(output)
       output = Cabin::Outputs::IO.new(output)
     end
     @subscriber_lock.synchronize do
-      @subscribers[output.object_id] = output
+      @subscribers[output.object_id] = [output, opts]
```

Comment:
Should we instead have a Subscription class instead of an array like this? Might make more sense from an API perspective.